### ISSUE TITLE
Table: a11y columnheaders

### ISSUE THREAD

**Issue by danethurber** [pluralsight/design-system#1024]
the current column headers when rendered as a button fail a11y checks because a button can't have a role of columnheader

also. we should change the event argument order of the `onClick` prop so that the click event is the first argument

related thread -> https://pluralsight.slack.com/archives/C70HPSJPP/p1591279249202900
